packages/babel-core/src/transform-ast.js

```diff
@@ -48,12 +44,6 @@ export const transformFromAst: TransformFromAst = (function transformFromAst(
     opts = undefined;
   }
 
-  // For backward-compat with Babel 6, we allow sync transformation when
-  // no callback is given. Will be dropped in some future Babel major version.
-  if (callback === undefined) {
-    return transformFromAstRunner.sync(ast, code, opts);
```

Comment:
Maybe we could throw a nice error message, like `Starting from Babel 8.0.0, the 'transformFromAst' function expects a callback. If you need to call it synchronously, please use 'transformFromAstSync' instead.`.

Reply:
How does this look?
```
  if (callback === undefined) {
    throw new Error(
         "Starting from Babel 8.0.0, the 'transformFromAst' function expects a callback. If you 
         need to call it synchronously, please use 'transformFromAstSync",
    );
  }
```
I'll push the changes once you approve.

Reply:
It looks good :+1: